### ISSUE TITLE
Docs docs docs

### ISSUE THREAD

**Issue by adelbertc** [typelevel/cats#1443]
In light of the new [docs](https://github.com/typelevel/cats/pull/1432) I think we should revisit our docs and update them to better reflect the new ordering. Going to use this ticket to track TODO's and in-flights.

- [ ] Type classes https://github.com/typelevel/cats/pull/1440
- [ ] Semigroup and Monoid https://github.com/typelevel/cats/pull/1442
- [ ] Traversable Functors
- [ ] Monads
- [ ] Variance and Functors

To be ordered/added/revised:
- [ ] Foldable
- [ ] Alternative/MonadCombine
- [ ] SemigroupK/MonoidK
- [ ] MTL (Monad{Reader, Writer, State, Error})

- [ ] Const
- [ ] Either/EitherT
- [ ] FreeApplicative
- [ ] Free
- [ ] Reader/ReaderT/Kleisli
- [ ] NonEmptyList
- [ ] NonEmptyVector
- [ ] Option/OptionT
- [ ] State/StateT
- [ ] Writer/WriterT
- [ ] Validated

**Comment by adelbertc** [comment]
I am working on the Traversable Functors section right now.

**Comment by adampingel** [comment]
If I can make a request of the docs here:

I was just reading through the pages for Free Monad and Free Applicative. I noticed this is the only place where the notion of an `algebra` is discussed. It would be helpful to either define it inline or link to the definition being used in those two docs.

**Comment by fommil** [comment]
I am forming some thoughts on the docs, but I'm not sure how long it will be before I have something constructive to add. Generally I feel many big problems are structural. e.g. getting to the docs (typelevel.org has a tendency to send people to github), ordering of entries, and the lack of an introduction / motivation.

It definitely assumes a lot of the reader. I think it can only be called a success when an OOP Java dev can read it after Odersky's book and get it.

Any time I read something an extraordinary intro to FP I document it here http://ensime.org/learning_scala/

**Comment by mgttlinger** [comment]
Imho the selling point to get the OO guys hooked is effect capturing structures like `Option` because everybody can relate to those pesky `NullPointerException`s. From there you can sell `Functor`, `Applicative`, `Monad` as a way to use those in a practical manner.

**Comment by fommil** [comment]
@mgttlinger I don't agree with that. It's simply not a pain point. The biggest selling point to OO developers, in my experience of talking to very sceptical / traditional developers, is being able to take the mocking pattern further and factor out async in their tests. Writing async tests is a massive PITA. OO devs get really annoyed with `Option` because it just means they have to do a `getOrElse(throw new ...)` everytime someone hands it to them.

**Comment by mgttlinger** [comment]
@fommil That's interesting because the guys at work got why `Option` etc is great rather quickly and use it but still resort to traditional mocking/testing stuff because they do not want to learn about `Freer` etc. But it may just be the team I work in. Who knows :man_shrugging:

**Comment by ChristopherDavenport** [comment]
I agree with @mgttlinger as far as the ways that I am able to convince people to start picking up FP paradigms for devs who come for OO backgrounds. We definitely need a more complete resource as @fommil suggests to advance people in practical as well as theoretical underpinnings.

As far as `getOrElse(throw new ...)` I have found that pain point also serves as a great transition into the power of ADT's, where I dont mention the scary sounding Co words, except with those who have already transitioned into FP. 

Regardless, if the docs are being spruced up making them more palatable to less-experienced developers is always the way to go. Even so far as warning which areas are more difficult if necessary.

**Comment by kailuowang** [comment]
From my experience, people coming to cats because they already want to do, or try, pure FP in scala. Even if they do not already know what FP is about, they at least have some preconceived notions about it. 
The sheer size of cats makes it rather intimidating to entice people to the FP world. I mean when you come to cats, where do you even start? typeclass? `Monad`? `Functor`? `Free`? 
There is already a good resource to gradually introduce new scala dev into the FP world, that's the red book. I don't see some paragraphs of cats documentation can achieve the same goal.
